src/AlertBanner/AlertBanner.tsx

```diff
@@ -0,0 +1,102 @@
+/** @jsx jsx */
+import { jsx } from "@emotion/core";
+import React, { CSSProperties, useMemo } from "react";
+import PropTypes from "prop-types";
+
+import { base } from "../typography";
+import { colors } from "../colors";
+import { assertUnreachable } from "../shared/assertUnreachable";
+import { IconInfoSolid } from "../icons/IconInfoSolid";
+import { IconWarningSolid } from "../icons/IconWarningSolid";
+import { IconErrorSolid } from "../icons/IconErrorSolid";
+import { IconSuccessSolid } from "../icons/IconSuccessSolid";
+
+interface AlertBannerProps {
+  /**
+   * color theme for alert
+   * @default "light"
+   */
+  theme?: "light" | "dark";
+
+  /**
+   * The content of the banner, appears to right of icon
+   */
+  children?: React.ReactNode;
+
+  className?: string;
+  style?: CSSProperties;
+
+  /**
+   * Type of alert, this is used to determine the color and icon in the banner
+   */
+  type: "info" | "warn" | "error" | "success";
+}
+
+export const AlertBanner: React.FC<AlertBannerProps> = ({
+  children,
+  theme = "light",
+  type,
+  ...otherProps
+}) => {
+  const { Icon, color: colorTemp } = useMemo(() => {
+    switch (type) {
+      case "info":
+        return { color: colors.blue, Icon: IconInfoSolid };
+      case "warn":
+        return { color: colors.orange, Icon: IconWarningSolid };
+      case "error":
+        return { color: colors.red, Icon: IconErrorSolid };
+      case "success":
+        return { color: colors.green, Icon: IconSuccessSolid };
+      default:
+        assertUnreachable(type);
+    }
+  }, [type]);
+  return (
+    <section
+      {...otherProps}
+      css={{
+        ...(theme === "light"
+          ? {
+              backgroundColor: colorTemp.lightest,
+              color: colorTemp.darker,
+              borderWidth: 1,
+              borderColor: colorTemp.lighter,
+            }
+          : theme === "dark"
+          ? {
+              backgroundColor: colorTemp.darkest,
+              color: colorTemp.lightest,
+              borderWidth: 0,
+              borderColor: colors.silver.dark,
+            }
+          : assertUnreachable(theme)),
+        boxShadow: `0 4px 8px 0 rgba(0, 0, 0, .04)`,
+        borderStyle: "solid",
+        borderRadius: 4,
+        padding: "12px 15px",
+        display: "flex",
+        alignItems: "center",
+        ...base.small,
+      }}
+    >
+      <Icon
+        css={{
+          width: 20,
+          height: 20,
+          color: colorTemp.base,
+          marginRight: 13,
+          "& .inner": theme === "dark" &&
+            type !== "warn" && { fill: colors.white },
```

Comment:
😭

Reply:
Yeah, the more I've thought about this the more it seems kind of ridiculous even from a design perspective. Maybe we can change the color value of the triangle to something that makes it work well with just white in either scenario. I kinda hate having that inconsistency too.